File: №3_4.cpp
#include <iostream>
using namespace std;

int main() {
    int n;
    cin >> n;
    
    for (int i = 0; i < n; i++) {
        string num;
        cin >> num;
        
        string rev = string(num.rbegin(), num.rend());
        cout << (num == rev ? "Yes " : "No ");
    }
    
    return 0;
}

File: №3_4.go
package main

import "fmt"

func main() {
	var n int
	fmt.Print("Введите N: ")
	fmt.Scan(&n)

	fmt.Print("Введите числа: ")
	for i := 0; i < n; i++ {
		var num string
		fmt.Scan(&num)

		rev := ""
		for j := len(num) - 1; j >= 0; j-- {
			rev += string(num[j])
		}

		if num == rev {
			fmt.Print("Yes ")
		} else {
			fmt.Print("No ")
		}
	}
	fmt.Println()
}
